### PR TITLE
feat: Add LiveQuery module to SDK

### PR DESCRIPTION
**Pull request by vazarkevych**
### Issue

Closes: https://github.com/parse-community/Parse-SDK-iOS-OSX/issues/1714

### TODOs before merging

- [ ] Added module ParseLiveQuery to SDK project
- [ ] Added SPM support for the module ParseLiveQuery
- [ ] Added build module ParseLiveQuery from CI/CD (this change needs to check because the test finished with an error. Locally everything works but on the repo we have an error)

### CODE REVIEW

**Comment by mtrezza** [comment]
@vazarkevych Could you give a description of the issues that are stopping this PR currently?

**Comment by vazarkevych** [comment]
Hi, @dplewis, Just a kind follow-up regarding the status of the ParseLiveQuery. Do you have any updates on your side?

**Comment by dplewis** [comment]
@vazarkevych I fixed the merge conflict waiting for CI to pass

**Comment by dplewis** [comment]
@vazarkevych I got the live query project to build successfully, looks like the release build is broken

**Comment by dplewis** [comment]
@vazarkevych @mtrezza The project builds and the CI is working. I don't know what else needs to be done.

There are some issue with running the project locally. I use Xcode 14 while the CI uses Xcode 11. I opened a PR on BoltsSwift https://github.com/BoltsFramework/Bolts-Swift/pull/97 which will help with migration to support Xcode 14.

**Comment by mtrezza** [comment]
@vazarkevych Could you look into the remaining issues so we can get this merged?

**Comment by vazarkevych** [comment]
Hi @dplewis, thank you for your contribution. I just checked SPM, and it stopped working

**Comment by dplewis** [comment]
@vazarkevych I don't think I changed anything SPM related.

**Comment by dplewis** [comment]
@vazarkevych i tried adding this package via  SPM and ran into https://github.com/parse-community/Parse-SDK-iOS-OSX/issues/1729

**Comment by dplewis** [comment]
@vazarkevych @mtrezza I got this to work via SPM. I added ParseObjC and ParseLiveQuery to my project from your github url and addParseLiveQuery branch. Then used

```
#import "Parse.h"
@import ParseLiveQuery;
```

**Review by dplewis — DISMISSED** [review]
Documentation, README, code coverage, starter projects, demo, and tests can be done in separate PRs. Also there are still PR's left on the LiveQuery repo that need to be addressed

**Comment by mtrezza** [comment]
Maybe we could at least copy/paste the LiveQuery README to this README under a new chapter?

**Comment by dplewis** [comment]
That documentation is outdated and very hard to get working for new users

**Comment by mtrezza** [comment]
Got it, then let's just wait for a review by @vazarkevych to confirm all is working as expected and then merge.

File: Package.swift
Comment (mtrezza) [inline Package.swift:13]: Why this change and is this a breaking change?

It seems a bit odd to name the core module `Parse`, as the term "Parse" is arguably referring to the whole SDK. To me is makes more sense to name the core module `ParseCore` to indicate that it's the basis that is always needed, while others are optional.

File: Cartfile.resolved
Comment (mtrezza) [inline Cartfile.resolved:2]: Do we really need 2 Bolts modules now? I assume they provide the same functionality, and if so, shouldn't the be usable interchangeably in ObjC and Swift code?

File: Package.swift
Comment (vazarkevych) [reply Package.swift:13]: Hi, yes it is an important change because in SDK target has the name "Parse". When we set "ParseCore" it causes an error downloading.

Comment (mtrezza) [reply Package.swift:13]: Not sure I understand. Why can't we keep the name `ParseCore` and change the target name? I mean this is only an issue for the LiveQuery module, right?

File: Cartfile.resolved
Comment (mtrezza) [reply Cartfile.resolved:2]: Got it, then let's keep both for now.

File: Package.swift
Comment (dplewis) [reply Package.swift:13]: @mtrezza This is my first time using SPM and I didn't know you had to do `import ParseCore;`. I can try to revert it back and fix the downloading error

Comment (mtrezza) [reply Package.swift:13]: @vazarkevych Could you take a look at this so we can close this PR? It seems to be the last remaining issue here.

Comment (VolodyaNazarkevych) [reply Package.swift:13]: @mtrezza Hi, This issue with renaming is caused because of we have in one project two code languages (swift, objc-c) and it has two different imports SDK to the project. If we left the target name ParseCore then we get the error with building ParseLiveQuery because the compiler doesn't find "ParseCore" because the target name in the project ParseSDK is "Parse", but SPM should work that SPM imports frameworks from "Package.swift" file. 

And, if we rename the target name from "ParseCore" to "Parse", we have successfully built the project ParseLiveQuery and have works SPM.

We saw two ways to resolve this issue:
First - Left the target name "Parse" (All our users will require to change the name of the import to "Parse" from "ParseCore" only in Swift, on Objective-c no changes are required).
Second - We need to rename the project "ParseSDK" targets from "Parse" to "ParseCore"

Comment (mtrezza) [reply Package.swift:13]: > Second - We need to rename the project "ParseSDK" targets from "Parse" to "ParseCore"

Let's see if I get this right. If we do this, then developers won't have a rename their `ParseCore` imports to `Parse`. If there are no other consequences then this is the preferred way.

Comment (project-academy) [reply Package.swift:13]: Any updates on this?

Comment (mtrezza) [reply Package.swift:13]: We just need someone to do this renaming, then we should be good to merge.

Comment (dplewis) [reply Package.swift:13]: @VolodyaNazarkevych @mtrezza  Either of you want to take this over?

Comment (mtrezza) [reply Package.swift:13]: It's up for grabs!

Comment (dplewis) [reply Package.swift:13]: @mtrezza Can you commit 1 example of tedious work? I'll finish the rest. I don't quite understand what needs to be changed

Comment (mtrezza) [reply Package.swift:13]: See https://github.com/parse-community/Parse-SDK-iOS-OSX/pull/1712#discussion_r1175802198.

Comment (project-academy) [reply Package.swift:13]: Any updates on this? 
Is this repo not really supported anymore? I'll adjust my expectations if so

Comment (mtrezza) [reply Package.swift:13]: You can see the status from the message above.

Comment (mtrezza) [reply Package.swift:13]: @dplewis Do you want to take a look at this, we could add an additional bounty.

Comment (mtrezza) [reply Package.swift:13]: @dplewis Do you want to take a look at this, we could add an additional bounty.

**Comment by dplewis** [comment]
@mtrezza @vazarkevych @VolodyaNazarkevych This is ready for review.

TODO: https://github.com/parse-community/Parse-SDK-iOS-OSX/pull/1712#pullrequestreview-1383664968

**Comment by mtrezza** [comment]
@VolodyaNazarkevych Could you please take a look at this?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [2.3.0](https://github.com/parse-community/Parse-SDK-iOS-OSX/releases/tag/2.3.0)

**Pull request merged**
